lib/syskit/roby_app/log_transfer_integration/log_transfer_integration.rb

```diff
@@ -0,0 +1,153 @@
+                        )
+                    )
+                    root_ca.add_extension(
+                        ext_fact.create_extension(
+                            "authorityKeyIdentifier", "keyid:always", true
+                        )
+                    )
+                end
+
+                # Inserts extra properties to the certificate creation process
+                def certificate_extension_add(ext_fact, cert)
+                    cert.add_extension(
+                        ext_fact.create_extension(
+                            "keyUsage", "digitalSignature", true
+                        )
+                    )
+                    cert.add_extension(
+                        ext_fact.create_extension(
+                            "subjectKeyIdentifier", "hash", false
+                        )
+                    )
+                end
+
+                # Creates a Root Certificate Authority from a root key
+                def create_root_ca(root_key)
+                    root_ca = OpenSSL::X509::Certificate.new
+                    hundred_years = 100 * 365 * 24 * 60 * 60
+                    pre_config(
+                        creation_cert: root_ca,
+                        key: root_key,
+                        subject: "/DC=org/DC=ruby-lang/CN=Ruby CA",
+                        issuer: root_ca,
+                        lifespan: hundred_years
+                    )
+                    ef_ca = OpenSSL::X509::ExtensionFactory.new
+                    ef_ca.subject_certificate = root_ca
+                    ef_ca.issuer_certificate = root_ca
+                    root_ca_extension_add(ef_ca, root_ca)
+                    root_ca.sign(root_key, OpenSSL::Digest::SHA256.new)
+                    root_ca
+                end
+
+                # Creates a Certificate authenticated by a Root CA
+                def create_cert(root_ca, root_key)
+                    key = OpenSSL::PKey::RSA.new 2048
+                    cert = OpenSSL::X509::Certificate.new
+                    single_year = 1 * 365 * 24 * 60 * 60
+                    pre_config(
+                        creation_cert: cert,
+                        key: key,
+                        subject: "/DC=org/DC=ruby-lang/CN=Ruby certificate",
+                        issuer: root_ca,
+                        lifespan: single_year
+                    )
+                    ef_cert = OpenSSL::X509::ExtensionFactory.new
+                    ef_cert.subject_certificate = cert
+                    ef_cert.issuer_certificate = root_ca
+                    certificate_extension_add(ef_cert, cert)
+                    cert.sign(root_key, OpenSSL::Digest::SHA256.new)
+                    cert
+                end
+
+                # Write created Certificate to file
+                def write_certificate(cert)
+                    cert_filepath = File.join(__dir__,
+                        "..", "log_transfer_integration", "cert.crt")
```

Comment:
If you want to write a certificate to a file, not passing the file path as parameter is rather obviously a Very Bad Idea(tm).

How does it work if I create two instances of TmpRootCA ?

Plus, writing in the tools/syskit package for an application is a VERY RED flag.